tests/neg/i5202.scala

```diff
@@ -0,0 +1,6 @@
+object Test {
+  val f: (Int => Int) | (String => Int) = (a: Int) => a + 3
+
+  f.apply(5) // error - found: Int expected: Int & String
+  f("c")     // error - found: String expected: Int & String
+}
```

Comment:
More examples would be nice, e.g. I had one at https://github.com/lampepfl/dotty/issues/5202#issuecomment-427058428, which could be adapted to try every combination of (covariant, invariant, invariant) x (and, or)